evaluator/evaluate.go

```diff
@@ -100,48 +102,54 @@ func commitWithObserve(tx *gorm.DB) error {
 }
 
 func evaluateAndStore(tx *gorm.DB, system *models.SystemPlatform, vmaasData vmaas.UpdatesV2Response) error {
-	patched, unpatched, err := processSystemAdvisories(tx, system, vmaasData, system.InventoryID)
+	oldSystemAdvisories, patched, unpatched, err := processSystemAdvisories(tx, system, vmaasData, system.InventoryID)
 	if err != nil {
 		evaluationCnt.WithLabelValues("error-process-advisories").Inc()
 		return errors.Wrap(err, "Unable to process system advisories")
 	}
 
-	err = storeAdvisoryData(tx, system, patched, unpatched)
+	newSystemAdvisories, err := storeAdvisoryData(tx, system, patched, unpatched)
 	if err != nil {
 		evaluationCnt.WithLabelValues("error-store-advisories").Inc()
 		return errors.Wrap(err, "Unable to store advisory data")
 	}
 
-	err = updateCaches(tx, system)
+	err = updateSystemPlatform(tx, system, oldSystemAdvisories, newSystemAdvisories)
 	if err != nil {
-		evaluationCnt.WithLabelValues("error-update-system-caches").Inc()
-		return errors.Wrap(err, "Unable to update system caches")
+		evaluationCnt.WithLabelValues("error-update-system").Inc()
+		return errors.Wrap(err, "Unable to update system")
 	}
 
-	err = updateSystemLastEvaluation(tx, system)
-	if err != nil {
-		evaluationCnt.WithLabelValues("error-update-last-eval").Inc()
-		return errors.Wrap(err, "Unable to update last_evaluation timestamp")
-	}
 	return nil
 }
 
-func updateSystemLastEvaluation(tx *gorm.DB, system *models.SystemPlatform) error {
+func updateSystemPlatform(tx *gorm.DB, system *models.SystemPlatform, old, new SystemAdvisoryMap) error {
 	tStart := time.Now()
-	defer utils.ObserveSecondsSince(tStart, evaluationPartDuration.WithLabelValues("last-evaluation-update"))
+	defer utils.ObserveSecondsSince(tStart, evaluationPartDuration.WithLabelValues("system-update"))
+	if old == nil || new == nil {
```

Comment:
I did not find the case where old or new could be nil. Is there any?

Reply:
Happened during some test cases.